### ISSUE TITLE
Troubleshooting guide for upgrade to 4.8.0

### ISSUE THREAD

**Issue by juliamagan** [wazuh/wazuh-documentation#7395]
## Description

Following the release of 4.8.0, it has been noticed in the community that multiple users are encountering the same problems when upgrading their environment to 4.8.0. Among them are:

- Changes in `opensearch_dashboard.yml `
- Vulnerability Detection configuration change in `ossec.conf`
- Credentials needed for the indexer in the keystore
- Increased resource usage

We need to create a troubleshooting guide with the most common errors to ease the upgrade process for the user. Also, we should check if the current upgrade or Vulnerability Detection documentation could be improved so that the user does not encounter these errors.

**Comment by GabrielEValenzuela** [comment]
# Comment

- This section could be improved, describing what `username` and `password` should be or for what.

```rst
Save the Wazuh indexer username and password into the Wazuh manager keystore using the Wazuh-keystore tool.


/var/ossec/bin/wazuh-keystore -f indexer -k username -v <INDEXER_USERNAME>
/var/ossec/bin/wazuh-keystore -f indexer -k password -v <INDEXER_PASSWORD>
```

**Comment by pereyra-m** [comment]
## Comment

I'd update the upgrade guide to include a note that there are breaking changes in the Dashboard configuration

https://documentation.wazuh.com/current/upgrade-guide/upgrading-central-components.html#upgrading-the-wazuh-dashboard

For example, the default app 
```
uiSettings.overrides.defaultRoute: /app/wz-home 
```

**Comment by MiguelazoDS** [comment]
Related to the indexer configuration I successfully helped some users by suggesting the use of curl command before configuring the indexer, we can add that verification since we have many notes across the documentation about the use of cli tools or commands to validate configurations.

This is the command I suggested 

`curl -u <user>:<pass> --cacert <path.pem> --cert <path-client.pem> --key <path-client-key.pem> -X GET "https://<IP>:9200/_cluster/health"`

**Comment by sebasfalcone** [comment]
## Upgrade guide
Missing note at the keystore step:
![image](https://github.com/wazuh/wazuh-documentation/assets/34063881/ed6c2144-bf50-4b92-b1e9-cad9964b8a0a)


We could replicate the same that is here 
![image](https://github.com/wazuh/wazuh-documentation/assets/34063881/426e1651-205a-4fd3-ac24-94705da76ddf)

**Comment by rauldpm** [comment]
This issue is for a troubleshooting guide, please open an issue for each bug or fix needed to apply to the documentation.

**Comment by rauldpm** [comment]
@pereyra-m about https://github.com/wazuh/wazuh-documentation/issues/7395#issuecomment-2165633614, this will be included (URL change in opensearch_dashboard.yml)

@MiguelazoDS Thanks, we will take it into account

**Comment by HQuest** [comment]
OVA installs uses different certificate names - not filebeat.pem/filebeat-key.pem, but wazuh-server.pem/wazuh-server-key.pem.
Also it seems the upgrade process adds the indexer tag at the ossec.conf bottom, so no need to add a second set of entry - but users should still adjust the cert names accordingly.
Finally, at the indexer host example, 0.0.0.0:9200 does not seems to work, but 127.0.0.1:9200 or the actual host IP:9200 do work.

**Comment by tsarquis88** [comment]
As @HQuest says, some users have added manually the `indexer` configuration block after reading the [documentation](https://documentation.wazuh.com/current/user-manual/capabilities/vulnerability-detection/configuring-scans.html). This leads to problems when upgrading Wazuh to v4.8 since this config is automatically added at the bottom of the config file.

It may be a good idea to add a note advising to check the existence of the block before adding a new one.

**Comment by rauldpm** [comment]
@tsarquis88 The capability documentation does not state to add the new block to the configuration, it informs that you can find it in the ossec.conf file

The upgrade guide has this note, that in fact, is not correct, as it must specify that the old vulnerability-detector config should be replaced by the new vulnerability-detection config

> Note
> 
> If the /var/ossec/etc/ossec.conf configuration file was modified, it will not be replaced by the upgrade. You will therefore have to add the settings of the new capabilities manually. More information can be found in [User manual](https://documentation.wazuh.com/current/user-manual/index.html).

This is something that should be fixed, but not added as a troubleshooting step, as it is part of the upgrade process, please open an issue as this needs to be clarified

In the same way, if the Wazuh manager is upgraded and the ossec.conf file is replaced, the block should not be added manually, but if the upgrade process does not update it, the blocks should be added, a reason for the config not being updated is that the configuration file has modifications (by the user) and the system does not update it

Please open another issue as this should be clarified better

**Comment by rauldpm** [comment]
@HQuest I understood your point, but certificates can have multiple nomenclatures as each user can set his own after generating them. The code block is an example and is the user's responsibility to ensure that the code block added is correct depending on your configuration or infrastructure, like what you mention about the IP value, in any case, as you can see, I asked my colleague to open new issues to clarify those things better, we truly appreciate you contribution

**Comment by HQuest** [comment]
> @tsarquis88 The capability documentation does not state to add the new block to the configuration, it informs that you can find it in the ossec.conf file
> 
> > The following configuration block shows a configuration example for the Vulnerability Detection module. You can find the Vulnerability Detection settings in the Wazuh server [configuration file](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/index.html) at /var/ossec/etc/ossec.conf.
I disagree with your assessment. The line prior to this reads
> The Vulnerability Detection module is enabled by default.
And yet, it is not enabled by default if you have upgraded your system. As only the indexer is added automatically but not the vulnerability detection module, when the entire block of two elements, vuln detection and indexer are part of the example, this leads to incorrect actions by the application managers.
> 
> The upgrade guide has this note, that in fact, is not correct, as it must specify that the old vulnerability-detector config should be replaced by the new vulnerability-detection config
> 
> > Note
> > If the /var/ossec/etc/ossec.conf configuration file was modified, it will not be replaced by the upgrade. You will therefore have to add the settings of the new capabilities manually. More information can be found in [User manual](https://documentation.wazuh.com/current/user-manual/index.html).
> 
> This is something that should be fixed, but not added as a troubleshooting step, as it is part of the upgrade process, please open an issue as this needs to be clarified
> 
> In the same way, if the Wazuh manager is upgraded and the ossec.conf file is replaced, the block should not be added manually, but if the upgrade process does not update it, the blocks should be added, a reason for the config not being updated is that the configuration file has modifications (by the user) and the system does not update it
Once again, it feels odd that one element is added when both are listed as part of the feature to be enabled.

**Comment by rauldpm** [comment]
@HQuest Yes the configuration when upgrading that you are mentioning is expected, and you need to follow the upgrade guide to replace the vulnerability block

>I disagree with your assessment. The line prior to this reads
>
>    The Vulnerability Detection module is enabled by default.

> And yet, it is not enabled by default if you have upgraded your system. As only the indexer is added automatically but not the vulnerability detection module, when the entire block of two elements, vuln detection and indexer are part of the example, this leads to incorrect actions by the application managers.

The capability documentation refers to Vulnerability `Detection`, not Vulnerability `Detector`, note that they are different modules, and the first is supposed to replace the second `which in a new installation is enabled by default, the old module is disabled by default`. You will need to add the new module manually and remove the old one if you want to use it (this is why I asked my colleague to open an issue and clarify it better)

The Vulnerability Detector module is `deprecated`, which does not mean non-functional

If you have an issue with the process or you encounter a bug, please feel free to open a new issue if the troubleshooting guide does not refer to your problem: https://documentation.wazuh.com/current/upgrade-guide/troubleshooting.html#none-of-the-above-solutions-are-fixing-my-problem
